### DISCUSSION TITLE
LINKERD2_PROXY_INBOUND_LISTENER_ADDR is reported to not be set

### DISCUSSION THREAD

**Discussion by alekhrycaiko** [linkerd/linkerd2#11941] [Help]
Hi, 

I'm wanting some clarity on what this log means from the destination container. I can't find any negative outcomes from this log in our system, but the frequency of the error has increased significantly following our upgrade to `stable-2.14.8`.

```
time="2024-01-16T09:22:34Z" level=error msg="failed to find LINKERD2_PROXY_INBOUND_LISTEN_ADDR environment variable in any container for given pod spec" addr=":8086" component=endpoint-profile-translator context-ns=<redacted> context-pod=<redacted> remote="<redacted>"
```

When investigating the redacted Pods, I do see that the LINKERD2_PROXY_INBOUND_LISTEN_ADDR environment variable is set. My current theory is that this is a race condition with the injector/destination port. Not super clear what would've changed in Linkerd2 to have caused this and/or whether there's something we can configure to remove this issue altogether.

Thanks in advance for any eye or help here 🙏

**Comment by mateiidavid** [comment]
@alekhrycaiko is the pod injected? And are you sending traffic to it directly (i.e. through its IP and not through a service object)?

**Reply by alekhrycaiko** [reply]
@mateiidavid Yes, the pod is injected. We do communication through service objects, not IP addresses.
